kore/src/Kore/Step/Simplification/And.hs

```diff
@@ -176,19 +189,50 @@ makeEvaluateNonBool
         initialConditions = firstCondition <> secondCondition
         merged = Conditional.andCondition terms initialConditions
     normalized <- Substitution.normalize merged
-    return (applyAndIdempotence <$> normalized)
-        { predicate = applyAndIdempotence <$> Conditional.predicate normalized }
-
-applyAndIdempotence
+    return
+        (applyAndIdempotenceAndFindContradictions <$> normalized)
+            { predicate =
+                applyAndIdempotenceAndFindContradictions
+                    <$> Conditional.predicate normalized
+            }
+
+applyAndIdempotenceAndFindContradictions
```

Comment:
Could you add tests for the "find contradictions" part?

Reply:
Done!